ops/testing.py

```diff
@@ -2586,6 +2606,118 @@ class NonAbsolutePathError(Exception):
     """
 
 
+class ContainerFilesystem(Protocol):
+    """The subset of container operations that are filesystem-related.
+
+    Don't instantiate this directly; use :meth:`Harness.get_container_filesystem`.
+    """
+
+    @abstractmethod
+    def pull(self, path: StrOrPath, *,
+             encoding: Optional[str] = 'utf-8') -> Union[BinaryIO, TextIO]:
+        """Read a file's content from the remote system.
+
+        See :meth:`model.Container.pull` for details.
```

Comment:
Sorry if I put you on the wrong track earlier, but these need the full `ops.model...` prefix to be linked correctly in the generated docs. Try `tox -e docs` and then `open docs/_build/html/index.html` to test it.

Reply:
Updated, I check the document and it's linked correctly now.